src/ServiceControl.ASB.DLQMonitor/CheckDeadLetterQueue.cs

```diff
@@ -0,0 +1,47 @@
+﻿namespace ServiceControl.ASB.DLQMonitor
+{
+    using System;
+    using System.Configuration;
+    using NServiceBus.CustomChecks;
+    using Microsoft.ServiceBus;
+    using ServiceBus.Management.Infrastructure.Settings;
+    using NServiceBus.Logging;
+
+    public class CheckDeadLetterQueue : CustomCheck
+    {
+        NamespaceManager namespaceManager;
+        string stagingQueue;
+
+        public CheckDeadLetterQueue(Settings settings) : base(id: "Dead Letter Queue", category: "Transport", repeatAfter: TimeSpan.FromHours(1))
+        {
+            Logger.Debug("ASB Dead Letter Queue custom check starting");
+
+            var connectionStringSettings = ConfigurationManager.ConnectionStrings["NServiceBus/Transport"];
+            var transportConnectionString = connectionStringSettings.ConnectionString;
+            namespaceManager = NamespaceManager.CreateFromConnectionString(transportConnectionString);
+            stagingQueue = $"{settings.ServiceName}.staging";
+        }
+
+
+        public override CheckResult PerformCheck()
+        {
+            Logger.Debug("Checking DLQ length");
```

Comment:
For [consistencey sake](https://github.com/Particular/ServiceControl/pull/1043/files#diff-3e47917f3d67b420746128d217e83f84R17), would replace DLQ with dead-letter queue.